File: ServiceExercise_2/app/src/main/java/com/programming/android/sdu/serviceexercise/MainActivity.java
package com.programming.android.sdu.serviceexercise;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;


public class MainActivity extends BaseActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();

        textSwitcher.setInAnimation(this, android.R.anim.slide_in_left);
        textSwitcher.setOutAnimation(this, android.R.anim.slide_out_right);

        Intent intent = new Intent(this, JokeAndroidService.class);
        startService(intent);

        findViewById(R.id.btnNext).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getBaseContext(), NextActivity.class);
                startActivity(i);
            }
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Intent intent = new Intent(this, JokeAndroidService.class);
        stopService(intent);
    }
}

File: Android 10 - Android services - executing long running tasks/ServiceExercise_2/app/src/main/java/com/programming/android/sdu/serviceexercise/MainActivity.kt
package com.programming.android.sdu.serviceexercise

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i("service_exrecise", "MainActivity onCreate- Current Thread ID- " + Thread.currentThread().id + " For Thread- " + Thread.currentThread().name)
        setContentView(R.layout.activity_main)
        initViews()
        textSwitcher.setInAnimation(this, android.R.anim.slide_in_left)
        textSwitcher.setOutAnimation(this, android.R.anim.slide_out_right)
        val btnNext: Button = findViewById(R.id.btnNext)
        btnNext.setOnClickListener {
            val i = Intent(baseContext, NextActivity::class.java)
            startActivity(i)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        val intent = Intent(this, JokeAndroidService::class.java)
        stopService(intent)
    }
}